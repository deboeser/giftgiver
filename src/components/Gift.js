import React, { Component } from "react";
import { Form } from "react-bootstrap";

class Gift extends Component {
  constructor() {
    super();

    this.state = {
      person: "",
      gift: ""
    };
  }

  onChange = e => {
    this.setState({
      [e.target.name]: e.target.value
    });
  };

  render() {
    return (
      <div>
        <Form>
          <Form.Group>
            <Form.Label>Person</Form.Label>
            <Form.Control
              className="input-person"
              name="person"
              onChange={e => this.onChange(e)}
            />
          </Form.Group>
          <Form.Group>
            <Form.Label>Gift</Form.Label>
            <Form.Control
              className="input-gift"
              name="gift"
              onChange={e => this.onChange(e)}
            />
          </Form.Group>
        </Form>
      </div>
    );
  }
}

export default Gift;
