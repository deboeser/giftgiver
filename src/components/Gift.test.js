import React from "react";
import { shallow } from "enzyme";
import Gift from "./Gift";

describe("Gift", () => {
  const gift = shallow(<Gift />);

  it("renders properly", () => {
    expect(gift).toMatchSnapshot();
  });

  it("initializes a person and gift in `state`", () => {
    expect(gift.state()).toEqual({ person: "", gift: "" });
  });

  describe("when typing into the `person` input", () => {
    const person = "uncle";

    beforeEach(() => {
      gift
        .find(".input-person")
        .simulate("change", { target: { value: person, name: "person" } });
    });

    it("updates the `person` in `state`", () => {
      expect(gift.state().person).toEqual(person);
    });
  });

  describe("when typing into the `gift` input", () => {
    const giftitem = "whiskey";

    beforeEach(() => {
      gift
        .find(".input-gift")
        .simulate("change", { target: { value: giftitem, name: "gift" } });
    });

    it("updates the `gift` in `state`", () => {
      expect(gift.state().gift).toEqual(giftitem);
    });
  });
});
