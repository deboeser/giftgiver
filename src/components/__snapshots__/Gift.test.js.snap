// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift renders properly 1`] = `
<div>
  <Form
    as="form"
    inline={false}
  >
    <FormGroup
      as="div"
    >
      <FormLabel
        column={false}
        srOnly={false}
      >
        Person
      </FormLabel>
      <FormControl
        as="input"
        className="input-person"
        name="person"
        onChange={[Function]}
      />
    </FormGroup>
    <FormGroup
      as="div"
    >
      <FormLabel
        column={false}
        srOnly={false}
      >
        Gift
      </FormLabel>
      <FormControl
        as="input"
        className="input-gift"
        name="gift"
        onChange={[Function]}
      />
    </FormGroup>
  </Form>
</div>
`;
